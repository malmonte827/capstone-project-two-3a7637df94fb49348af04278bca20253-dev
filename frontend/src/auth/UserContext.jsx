import React from "react"

/** COntext provides current user object throughout app */

const UserContext = React.createContext()

export default UserContext