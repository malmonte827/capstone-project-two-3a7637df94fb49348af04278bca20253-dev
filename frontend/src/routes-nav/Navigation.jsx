import React, { useContext } from "react";
import UserContext from "../auth/UserContext";
import { Link, NavLink } from "react-router-dom";
import "./Navigation.css"

/** Navigation Component
 *
 * Displays the sites navigation bar on every page
 *
 * when a user is logged in it shows links to other areas of site
 * when not shows links to signup and login forms
 *
 */
function Navigation({ logout }) {
    const { currentUser } = useContext(UserContext);

 
 // Renders the navigation links for logged-in users
    function loggedInNav() {
        return (
            <ul className="navbar-nav ms-auto">
                <li className="nav-item me-4">
                    <NavLink className="nav-link" to="/pets">
                        Pets
                    </NavLink>
                </li>
                <li className="nav-item me-4">
                    <NavLink className="nav-link" to="/profile">
                        Profile
                    </NavLink>
                </li>
                <li className="nav-item me-4">
                    <NavLink className="nav-link" to="/" onClick={logout}>
                        Logout {currentUser.firstName || currentUser.username}
                    </NavLink>
                </li>
            </ul>
        );
    }
 // Renders the navigation links for logged-out users

    function loggedOutNav() {
        return (
            <ul className="navbar-nav ms-auto">
                <li className="nav-item me-4">
                    <NavLink className="nav-link" to="/login">
                        Login
                    </NavLink>
                </li>
                <li className="nav-item me-4">
                    <NavLink className="nav-link" to="/signup">
                        Signup
                    </NavLink>
                </li>
            </ul>
        );
    }

    return (
        <nav className="Navigation navbar navbar-expandd-md">
            <div className="container-fluid">
                <Link className="navbar-brand" to="/">
                    Capstone
                </Link>
                {currentUser ? loggedInNav() : loggedOutNav()}
            </div>
        </nav>
    );
}

export default Navigation;
