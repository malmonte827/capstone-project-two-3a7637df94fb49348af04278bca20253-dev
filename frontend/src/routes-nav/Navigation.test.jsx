import React from "react";
import Navigation from "./Navigation";
import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserContext from "../auth/UserContext";
import { vi } from "vitest";

describe("<Navigation />", () => {
    const logout = vi.fn();

    describe("When the user is logged in", () => {
        const currentUser = { firstName: "John", username: "john_doe" };

        it("renders without crashing", () => {
            render(
                <MemoryRouter>
                    <UserContext.Provider value={{ currentUser }}>
                        <Navigation />
                    </UserContext.Provider>
                </MemoryRouter>
            );
        });

        it("matches snapshot", () => {
            const currentUser = { firstName: "John", username: "john_doe" };
            const { asFragment } = render(
                <MemoryRouter>
                    <UserContext.Provider value={{ currentUser }}>
                        <Navigation logout={logout} />
                    </UserContext.Provider>
                </MemoryRouter>
            );
            expect(asFragment()).toMatchSnapshot();
        });

        it("renders the logged-in navigation links", () => {
            render(
                <MemoryRouter>
                    <UserContext.Provider value={{ currentUser }}>
                        <Navigation logout={logout} />
                    </UserContext.Provider>
                </MemoryRouter>
            );

            expect(screen.getByText("Pets")).toBeInTheDocument();
            expect(screen.getByText("Profile")).toBeInTheDocument();
            expect(screen.getByText(`Logout John`)).toBeInTheDocument();
        });

        it("calls logout when the logout link is clicked", () => {
            render(
                <MemoryRouter>
                    <UserContext.Provider value={{ currentUser }}>
                        <Navigation logout={logout} />
                    </UserContext.Provider>
                </MemoryRouter>
            );

            const logoutLink = screen.getByText(`Logout John`);
            fireEvent.click(logoutLink);

            expect(logout).toHaveBeenCalled();
        });
    });

    describe("When the user is logged out", () => {
        it("renders the logged-out navigation links", () => {
            render(
                <MemoryRouter>
                    <UserContext.Provider value={{ currentUser: null }}>
                        <Navigation logout={logout} />
                    </UserContext.Provider>
                </MemoryRouter>
            );

            expect(screen.getByText("Login")).toBeInTheDocument();
            expect(screen.getByText("Signup")).toBeInTheDocument();
        });
    });
});
