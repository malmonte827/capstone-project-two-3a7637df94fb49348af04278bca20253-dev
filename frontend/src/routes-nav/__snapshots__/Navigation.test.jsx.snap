// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Navigation /> > When the user is logged in > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expandd-md"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        data-discover="true"
        href="/"
      >
        Capstone
      </a>
      <ul
        class="navbar-nav ms-auto"
      >
        <li
          class="nav-item me-4"
        >
          <a
            class="nav-link"
            data-discover="true"
            href="/pets"
          >
            Pets
          </a>
        </li>
        <li
          class="nav-item me-4"
        >
          <a
            class="nav-link"
            data-discover="true"
            href="/profile"
          >
            Profile
          </a>
        </li>
        <li
          class="nav-item me-4"
        >
          <a
            aria-current="page"
            class="nav-link active"
            data-discover="true"
            href="/"
          >
            Logout John
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
